underthesea-core 3.3.1 build errors

```bash
1/1) Arming ConditionNeedsUpdate...
==> Retrieving sources...
  -> Found underthesea_core-3.3.1.tar.gz
==> WARNING: Skipping all source file integrity checks.
==> Extracting sources...
  -> Extracting underthesea_core-3.3.1.tar.gz with bsdtar
==> Starting build()...
* Getting build dependencies for wheel...
* Building wheel...
Running `maturin pep517 build-wheel -i /usr/bin/python --compatibility off`
🐍 Found CPython 3.14 at /usr/bin/python
🔗 Found pyo3 bindings
📡 Using build options features from pyproject.toml
   Compiling target-lexicon v0.13.5
   Compiling once_cell v1.21.4
   Compiling proc-macro2 v1.0.106
   Compiling unicode-ident v1.0.24
   Compiling quote v1.0.45
   Compiling libc v0.2.186
   Compiling memchr v2.8.0
   Compiling crossbeam-utils v0.8.21
   Compiling zerocopy v0.8.48
   Compiling serde_core v1.0.228
   Compiling autocfg v1.5.1
   Compiling serde v1.0.228
   Compiling cfg-if v1.0.4
   Compiling pyo3-build-config v0.25.1
   Compiling crossbeam-epoch v0.9.18
   Compiling syn v2.0.117
   Compiling memoffset v0.9.1
   Compiling getrandom v0.2.17
   Compiling heck v0.5.0
   Compiling anyhow v1.0.102
   Compiling rayon-core v1.13.0
   Compiling rand_core v0.6.4
   Compiling crossbeam-deque v0.8.6
   Compiling aho-corasick v1.1.4
   Compiling bstr v1.12.1
   Compiling array-init v2.1.0
error[E0583]: file not found for module `bstring`
   --> /build/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/bstr-1.12.1/src/lib.rs:429:1
    |
429 | mod bstring;
    | ^^^^^^^^^^^^
    |
    = help: to create the module `bstring`, create file "/build/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/bstr-1.12.1/src/bstring.rs" or "/build/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/bstr-1.12.1/src/bstring/mod.rs"
    = note: if there is a `mod bstring` elsewhere in the crate already, import it with `use crate::...` instead

   Compiling jhash v0.1.1
error[E0583]: file not found for module `ext_vec`
   --> /build/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/bstr-1.12.1/src/lib.rs:434:1
    |
434 | mod ext_vec;
    | ^^^^^^^^^^^^
    |
    = help: to create the module `ext_vec`, create file "/build/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/bstr-1.12.1/src/ext_vec.rs" or "/build/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/bstr-1.12.1/src/ext_vec/mod.rs"
    = note: if there is a `mod ext_vec` elsewhere in the crate already, import it with `use crate::...` instead

error[E0583]: file not found for module `io`
   --> /build/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/bstr-1.12.1/src/lib.rs:437:1
    |
437 | pub mod io;
    | ^^^^^^^^^^^
    |
    = help: to create the module `io`, create file "/build/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/bstr-1.12.1/src/io.rs" or "/build/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/bstr-1.12.1/src/io/mod.rs"
    = note: if there is a `mod io` elsewhere in the crate already, import it with `use crate::...` instead

   Compiling lazy_static v1.5.0
   Compiling regex-automata v0.1.10
error[E0599]: no method named `push_str` found for mutable reference `&mut Vec<u8>` in the current scope
    --> /build/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/bstr-1.12.1/src/ext_slice.rs:1660:18
     |
1660 |             dest.push_str(&self.as_bytes()[last..start]);
     |                  ^^^^^^^^
     |
help: there is a method `push` with a similar name
     |
1660 -             dest.push_str(&self.as_bytes()[last..start]);
1660 +             dest.push(&self.as_bytes()[last..start]);
     |

error[E0599]: no method named `push_str` found for mutable reference `&mut Vec<u8>` in the current scope
    --> /build/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/bstr-1.12.1/src/ext_slice.rs:1661:18
     |
1661 |             dest.push_str(replacement);
     |                  ^^^^^^^^
     |
help: there is a method `push` with a similar name
     |
1661 -             dest.push_str(replacement);
1661 +             dest.push(replacement);
     |

error[E0599]: no method named `push_str` found for mutable reference `&mut Vec<u8>` in the current scope
    --> /build/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/bstr-1.12.1/src/ext_slice.rs:1664:14
     |
1664 |         dest.push_str(&self.as_bytes()[last..]);
     |              ^^^^^^^^
     |
help: there is a method `push` with a similar name
     |
1664 -         dest.push_str(&self.as_bytes()[last..]);
1664 +         dest.push(&self.as_bytes()[last..]);
     |

   Compiling tinyvec_macros v0.1.1
error[E0599]: no method named `push_str` found for mutable reference `&mut Vec<u8>` in the current scope
    --> /build/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/bstr-1.12.1/src/ext_slice.rs:1725:18
     |
1725 |             dest.push_str(&self.as_bytes()[last..start]);
     |                  ^^^^^^^^
     |
help: there is a method `push` with a similar name
     |
1725 -             dest.push_str(&self.as_bytes()[last..start]);
1725 +             dest.push(&self.as_bytes()[last..start]);
     |

error[E0599]: no method named `push_str` found for mutable reference `&mut Vec<u8>` in the current scope
    --> /build/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/bstr-1.12.1/src/ext_slice.rs:1726:18
     |
1726 |             dest.push_str(replacement);
     |                  ^^^^^^^^
     |
help: there is a method `push` with a similar name
     |
1726 -             dest.push_str(replacement);
1726 +             dest.push(replacement);
     |

error[E0599]: no method named `push_str` found for mutable reference `&mut Vec<u8>` in the current scope
    --> /build/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/bstr-1.12.1/src/ext_slice.rs:1729:14
     |
1729 |         dest.push_str(&self.as_bytes()[last..]);
     |              ^^^^^^^^
     |
help: there is a method `push` with a similar name
     |
1729 -         dest.push_str(&self.as_bytes()[last..]);
1729 +         dest.push(&self.as_bytes()[last..]);
     |

error[E0599]: no method named `push_str` found for struct `Vec<_>` in the current scope
   --> /build/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/bstr-1.12.1/src/impls.rs:355:21
    |
355 |                 buf.push_str(b);
    |                     ^^^^^^^^
    |
help: there is a method `push` with a similar name
    |
355 -                 buf.push_str(b);
355 +                 buf.push(b);
    |

error[E0599]: no method named `push_str` found for struct `Vec<_>` in the current scope
   --> /build/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/bstr-1.12.1/src/impls.rs:366:21
    |
366 |                 buf.push_str(b);
    |                     ^^^^^^^^
    |
help: there is a method `push` with a similar name
    |
366 -                 buf.push_str(b);
366 +                 buf.push(b);
    |

   Compiling bitflags v1.3.2
error[E0599]: no method named `push_str` found for struct `Vec<_>` in the current scope
   --> /build/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/bstr-1.12.1/src/impls.rs:377:21
    |
377 |                 buf.push_str(b);
    |                     ^^^^^^^^
    |
help: there is a method `push` with a similar name
    |
377 -                 buf.push_str(b);
377 +                 buf.push(b);
    |

error[E0599]: no method named `push_str` found for struct `Vec<_>` in the current scope
   --> /build/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/bstr-1.12.1/src/impls.rs:388:21
    |
388 |                 buf.push_str(b);
    |                     ^^^^^^^^
    |
help: there is a method `push` with a similar name
    |
388 -                 buf.push_str(b);
388 +                 buf.push(b);
    |

error[E0308]: mismatched types
   --> /build/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/bstr-1.12.1/src/impls.rs:896:54
    |
896 |             PartialOrd::partial_cmp(self.as_bytes(), other.as_bytes())
    |             -----------------------                  ^^^^^^^^^^^^^^^^ expected `&BStr`, found `&[u8]`
    |             |
    |             arguments to this function are incorrect
    |
    = note: expected reference `&BStr`
               found reference `&[u8]`
note: method defined here
   --> library/core/src/cmp.rs:1395:7
help: try removing the method call
    |
896 -             PartialOrd::partial_cmp(self.as_bytes(), other.as_bytes())
896 +             PartialOrd::partial_cmp(self.as_bytes(), other)
    |

error[E0308]: mismatched types
   --> /build/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/bstr-1.12.1/src/impls.rs:68:58
    |
 68 |                 PartialOrd::partial_cmp(self.as_bytes(), other)
    |                 -----------------------                  ^^^^^ expected `&BStr`, found `&[u8]`
    |                 |
    |                 arguments to this function are incorrect
...
907 |     impl_partial_ord!(BStr, [u8]);
    |     ----------------------------- in this macro invocation
    |
    = note: expected reference `&BStr`
               found reference `&[u8]`
note: method defined here
   --> library/core/src/cmp.rs:1395:7
    = note: this error originates in the macro `impl_partial_ord` (in Nightly builds, run with -Z macro-backtrace for more info)
help: call `Into::into` on this expression to convert `&[u8]` into `&BStr`
    |
 68 |                 PartialOrd::partial_cmp(self.as_bytes(), other.into())
    |                                                               +++++++

error[E0308]: mismatched types
   --> /build/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/bstr-1.12.1/src/impls.rs:76:47
    |
 76 |                 PartialOrd::partial_cmp(this, other.as_bytes())
    |                 -----------------------       ^^^^^^^^^^^^^^^^ expected `&BStr`, found `&[u8]`
    |                 |
    |                 arguments to this function are incorrect
...
907 |     impl_partial_ord!(BStr, [u8]);
    |     ----------------------------- in this macro invocation
    |
    = note: expected reference `&BStr`
               found reference `&[u8]`
note: method defined here
   --> library/core/src/cmp.rs:1395:7
    = note: this error originates in the macro `impl_partial_ord` (in Nightly builds, run with -Z macro-backtrace for more info)
help: try removing the method call
    |
 76 -                 PartialOrd::partial_cmp(this, other.as_bytes())
 76 +                 PartialOrd::partial_cmp(this, other)
    |

error[E0308]: mismatched types
   --> /build/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/bstr-1.12.1/src/impls.rs:68:58
    |
 68 |                 PartialOrd::partial_cmp(self.as_bytes(), other)
    |                 -----------------------                  ^^^^^ expected `&BStr`, found `&[u8]`
    |                 |
    |                 arguments to this function are incorrect
...
908 |     impl_partial_ord!(BStr, &'a [u8]);
    |     --------------------------------- in this macro invocation
    |
    = note: expected reference `&BStr`
               found reference `&[u8]`
note: method defined here
   --> library/core/src/cmp.rs:1395:7
    = note: this error originates in the macro `impl_partial_ord` (in Nightly builds, run with -Z macro-backtrace for more info)
help: call `Into::into` on this expression to convert `&[u8]` into `&BStr`
    |
 68 |                 PartialOrd::partial_cmp(self.as_bytes(), other.into())
    |                                                               +++++++

error[E0308]: mismatched types
   --> /build/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/bstr-1.12.1/src/impls.rs:76:47
    |
 76 |                 PartialOrd::partial_cmp(this, other.as_bytes())
    |                 -----------------------       ^^^^^^^^^^^^^^^^ expected `&BStr`, found `&[u8]`
    |                 |
    |                 arguments to this function are incorrect
...
908 |     impl_partial_ord!(BStr, &'a [u8]);
    |     --------------------------------- in this macro invocation
    |
    = note: expected reference `&BStr`
               found reference `&[u8]`
note: method defined here
   --> library/core/src/cmp.rs:1395:7
    = note: this error originates in the macro `impl_partial_ord` (in Nightly builds, run with -Z macro-backtrace for more info)
help: try removing the method call
    |
 76 -                 PartialOrd::partial_cmp(this, other.as_bytes())
 76 +                 PartialOrd::partial_cmp(this, other)
    |

error[E0308]: mismatched types
   --> /build/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/bstr-1.12.1/src/impls.rs:68:58
    |
 68 |                 PartialOrd::partial_cmp(self.as_bytes(), other)
    |                 -----------------------                  ^^^^^ expected `&BStr`, found `&[u8]`
    |                 |
    |                 arguments to this function are incorrect
...
909 |     impl_partial_ord!(BStr, str);
    |     ---------------------------- in this macro invocation
    |
    = note: expected reference `&BStr`
               found reference `&[u8]`
note: method defined here
   --> library/core/src/cmp.rs:1395:7
    = note: this error originates in the macro `impl_partial_ord` (in Nightly builds, run with -Z macro-backtrace for more info)
help: call `Into::into` on this expression to convert `&[u8]` into `&BStr`
    |
 68 |                 PartialOrd::partial_cmp(self.as_bytes(), other.into())
    |                                                               +++++++

error[E0308]: mismatched types
   --> /build/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/bstr-1.12.1/src/impls.rs:76:47
    |
 76 |                 PartialOrd::partial_cmp(this, other.as_bytes())
    |                 -----------------------       ^^^^^^^^^^^^^^^^ expected `&BStr`, found `&[u8]`
    |                 |
    |                 arguments to this function are incorrect
...
909 |     impl_partial_ord!(BStr, str);
    |     ---------------------------- in this macro invocation
    |
    = note: expected reference `&BStr`
               found reference `&[u8]`
note: method defined here
   --> library/core/src/cmp.rs:1395:7
    = note: this error originates in the macro `impl_partial_ord` (in Nightly builds, run with -Z macro-backtrace for more info)
help: try removing the method call
    |
 76 -                 PartialOrd::partial_cmp(this, other.as_bytes())
 76 +                 PartialOrd::partial_cmp(this, other)
    |

error[E0308]: mismatched types
   --> /build/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/bstr-1.12.1/src/impls.rs:68:58
    |
 68 |                 PartialOrd::partial_cmp(self.as_bytes(), other)
    |                 -----------------------                  ^^^^^ expected `&BStr`, found `&[u8]`
    |                 |
    |                 arguments to this function are incorrect
...
910 |     impl_partial_ord!(BStr, &'a str);
    |     -------------------------------- in this macro invocation
    |
    = note: expected reference `&BStr`
               found reference `&[u8]`
note: method defined here
   --> library/core/src/cmp.rs:1395:7
    = note: this error originates in the macro `impl_partial_ord` (in Nightly builds, run with -Z macro-backtrace for more info)
help: call `Into::into` on this expression to convert `&[u8]` into `&BStr`
    |
 68 |                 PartialOrd::partial_cmp(self.as_bytes(), other.into())
    |                                                               +++++++

error[E0308]: mismatched types
   --> /build/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/bstr-1.12.1/src/impls.rs:76:47
    |
 76 |                 PartialOrd::partial_cmp(this, other.as_bytes())
    |                 -----------------------       ^^^^^^^^^^^^^^^^ expected `&BStr`, found `&[u8]`
    |                 |
    |                 arguments to this function are incorrect
...
910 |     impl_partial_ord!(BStr, &'a str);
    |     -------------------------------- in this macro invocation
    |
    = note: expected reference `&BStr`
               found reference `&[u8]`
note: method defined here
   --> library/core/src/cmp.rs:1395:7
    = note: this error originates in the macro `impl_partial_ord` (in Nightly builds, run with -Z macro-backtrace for more info)
help: try removing the method call
    |
 76 -                 PartialOrd::partial_cmp(this, other.as_bytes())
 76 +                 PartialOrd::partial_cmp(this, other)
    |

error[E0308]: mismatched types
   --> /build/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/bstr-1.12.1/src/impls.rs:88:58
    |
 88 |                 PartialOrd::partial_cmp(self.as_bytes(), other)
    |                 -----------------------                  ^^^^^ expected `&BStr`, found `&[u8]`
    |                 |
    |                 arguments to this function are incorrect
...
911 |     impl_partial_ord_n!(BStr, [u8; N]);
    |     ---------------------------------- in this macro invocation
    |
    = note: expected reference `&BStr`
               found reference `&[u8]`
note: method defined here
   --> library/core/src/cmp.rs:1395:7
    = note: this error originates in the macro `impl_partial_ord_n` (in Nightly builds, run with -Z macro-backtrace for more info)
help: call `Into::into` on this expression to convert `&[u8]` into `&BStr`
    |
 88 |                 PartialOrd::partial_cmp(self.as_bytes(), other.into())
    |                                                               +++++++

error[E0308]: mismatched types
   --> /build/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/bstr-1.12.1/src/impls.rs:96:47
    |
 96 |                 PartialOrd::partial_cmp(this, other.as_bytes())
    |                 -----------------------       ^^^^^^^^^^^^^^^^ expected `&BStr`, found `&[u8]`
    |                 |
    |                 arguments to this function are incorrect
...
911 |     impl_partial_ord_n!(BStr, [u8; N]);
    |     ---------------------------------- in this macro invocation
    |
    = note: expected reference `&BStr`
               found reference `&[u8]`
note: method defined here
   --> library/core/src/cmp.rs:1395:7
    = note: this error originates in the macro `impl_partial_ord_n` (in Nightly builds, run with -Z macro-backtrace for more info)
help: try removing the method call
    |
 96 -                 PartialOrd::partial_cmp(this, other.as_bytes())
 96 +                 PartialOrd::partial_cmp(this, other)
    |

error[E0308]: mismatched types
   --> /build/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/bstr-1.12.1/src/impls.rs:88:58
    |
 88 |                 PartialOrd::partial_cmp(self.as_bytes(), other)
    |                 -----------------------                  ^^^^^ expected `&BStr`, found `&[u8]`
    |                 |
    |                 arguments to this function are incorrect
...
912 |     impl_partial_ord_n!(BStr, &'a [u8; N]);
    |     -------------------------------------- in this macro invocation
    |
    = note: expected reference `&BStr`
               found reference `&[u8]`
note: method defined here
   --> library/core/src/cmp.rs:1395:7
    = note: this error originates in the macro `impl_partial_ord_n` (in Nightly builds, run with -Z macro-backtrace for more info)
help: call `Into::into` on this expression to convert `&[u8]` into `&BStr`
    |
 88 |                 PartialOrd::partial_cmp(self.as_bytes(), other.into())
    |                                                               +++++++

error[E0308]: mismatched types
   --> /build/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/bstr-1.12.1/src/impls.rs:96:47
    |
 96 |                 PartialOrd::partial_cmp(this, other.as_bytes())
    |                 -----------------------       ^^^^^^^^^^^^^^^^ expected `&BStr`, found `&[u8]`
    |                 |
    |                 arguments to this function are incorrect
...
912 |     impl_partial_ord_n!(BStr, &'a [u8; N]);
    |     -------------------------------------- in this macro invocation
    |
    = note: expected reference `&BStr`
               found reference `&[u8]`
note: method defined here
   --> library/core/src/cmp.rs:1395:7
    = note: this error originates in the macro `impl_partial_ord_n` (in Nightly builds, run with -Z macro-backtrace for more info)
help: try removing the method call
    |
 96 -                 PartialOrd::partial_cmp(this, other.as_bytes())
 96 +                 PartialOrd::partial_cmp(this, other)
    |

error[E0308]: mismatched types
   --> /build/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/bstr-1.12.1/src/impls.rs:68:58
    |
 68 |                 PartialOrd::partial_cmp(self.as_bytes(), other)
    |                 -----------------------                  ^^^^^ expected `&BStr`, found `&[u8]`
    |                 |
    |                 arguments to this function are incorrect
...
915 |     impl_partial_ord!(BStr, Vec<u8>);
    |     -------------------------------- in this macro invocation
    |
    = note: expected reference `&BStr`
               found reference `&[u8]`
note: method defined here
   --> library/core/src/cmp.rs:1395:7
    = note: this error originates in the macro `impl_partial_ord` (in Nightly builds, run with -Z macro-backtrace for more info)
help: call `Into::into` on this expression to convert `&[u8]` into `&BStr`
    |
 68 |                 PartialOrd::partial_cmp(self.as_bytes(), other.into())
    |                                                               +++++++

error[E0308]: mismatched types
   --> /build/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/bstr-1.12.1/src/impls.rs:76:47
    |
 76 |                 PartialOrd::partial_cmp(this, other.as_bytes())
    |                 -----------------------       ^^^^^^^^^^^^^^^^ expected `&BStr`, found `&[u8]`
    |                 |
    |                 arguments to this function are incorrect
...
915 |     impl_partial_ord!(BStr, Vec<u8>);
    |     -------------------------------- in this macro invocation
    |
    = note: expected reference `&BStr`
               found reference `&[u8]`
note: method defined here
   --> library/core/src/cmp.rs:1395:7
    = note: this error originates in the macro `impl_partial_ord` (in Nightly builds, run with -Z macro-backtrace for more info)
help: try removing the method call
    |
 76 -                 PartialOrd::partial_cmp(this, other.as_bytes())
 76 +                 PartialOrd::partial_cmp(this, other)
    |

error[E0308]: mismatched types
   --> /build/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/bstr-1.12.1/src/impls.rs:68:58
    |
 68 |                 PartialOrd::partial_cmp(self.as_bytes(), other)
    |                 -----------------------                  ^^^^^ expected `&BStr`, found `&[u8]`
    |                 |
    |                 arguments to this function are incorrect
...
917 |     impl_partial_ord!(&'a BStr, Vec<u8>);
    |     ------------------------------------ in this macro invocation
    |
    = note: expected reference `&BStr`
               found reference `&[u8]`
note: method defined here
   --> library/core/src/cmp.rs:1395:7
    = note: this error originates in the macro `impl_partial_ord` (in Nightly builds, run with -Z macro-backtrace for more info)
help: call `Into::into` on this expression to convert `&[u8]` into `&BStr`
    |
 68 |                 PartialOrd::partial_cmp(self.as_bytes(), other.into())
    |                                                               +++++++

   Compiling regex-syntax v0.8.10
error[E0308]: mismatched types
   --> /build/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/bstr-1.12.1/src/impls.rs:76:47
    |
 76 |                 PartialOrd::partial_cmp(this, other.as_bytes())
    |                 -----------------------       ^^^^^^^^^^^^^^^^ expected `&BStr`, found `&[u8]`
    |                 |
    |                 arguments to this function are incorrect
...
917 |     impl_partial_ord!(&'a BStr, Vec<u8>);
    |     ------------------------------------ in this macro invocation
    |
    = note: expected reference `&BStr`
               found reference `&[u8]`
note: method defined here
   --> library/core/src/cmp.rs:1395:7
    = note: this error originates in the macro `impl_partial_ord` (in Nightly builds, run with -Z macro-backtrace for more info)
help: try removing the method call
    |
 76 -                 PartialOrd::partial_cmp(this, other.as_bytes())
 76 +                 PartialOrd::partial_cmp(this, other)
    |

error[E0308]: mismatched types
   --> /build/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/bstr-1.12.1/src/impls.rs:68:58
    |
 68 |                 PartialOrd::partial_cmp(self.as_bytes(), other)
    |                 -----------------------                  ^^^^^ expected `&BStr`, found `&[u8]`
    |                 |
    |                 arguments to this function are incorrect
...
919 |     impl_partial_ord!(BStr, String);
    |     ------------------------------- in this macro invocation
    |
    = note: expected reference `&BStr`
               found reference `&[u8]`
note: method defined here
   --> library/core/src/cmp.rs:1395:7
    = note: this error originates in the macro `impl_partial_ord` (in Nightly builds, run with -Z macro-backtrace for more info)
help: call `Into::into` on this expression to convert `&[u8]` into `&BStr`
    |
 68 |                 PartialOrd::partial_cmp(self.as_bytes(), other.into())
    |                                                               +++++++

error[E0308]: mismatched types
   --> /build/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/bstr-1.12.1/src/impls.rs:76:47
    |
 76 |                 PartialOrd::partial_cmp(this, other.as_bytes())
    |                 -----------------------       ^^^^^^^^^^^^^^^^ expected `&BStr`, found `&[u8]`
    |                 |
    |                 arguments to this function are incorrect
...
919 |     impl_partial_ord!(BStr, String);
    |     ------------------------------- in this macro invocation
    |
    = note: expected reference `&BStr`
               found reference `&[u8]`
note: method defined here
   --> library/core/src/cmp.rs:1395:7
    = note: this error originates in the macro `impl_partial_ord` (in Nightly builds, run with -Z macro-backtrace for more info)
help: try removing the method call
    |
 76 -                 PartialOrd::partial_cmp(this, other.as_bytes())
 76 +                 PartialOrd::partial_cmp(this, other)
    |

error[E0308]: mismatched types
   --> /build/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/bstr-1.12.1/src/impls.rs:68:58
    |
 68 |                 PartialOrd::partial_cmp(self.as_bytes(), other)
    |                 -----------------------                  ^^^^^ expected `&BStr`, found `&[u8]`
    |                 |
    |                 arguments to this function are incorrect
...
921 |     impl_partial_ord!(&'a BStr, String);
    |     ----------------------------------- in this macro invocation
    |
    = note: expected reference `&BStr`
               found reference `&[u8]`
note: method defined here
   --> library/core/src/cmp.rs:1395:7
    = note: this error originates in the macro `impl_partial_ord` (in Nightly builds, run with -Z macro-backtrace for more info)
help: call `Into::into` on this expression to convert `&[u8]` into `&BStr`
    |
 68 |                 PartialOrd::partial_cmp(self.as_bytes(), other.into())
    |                                                               +++++++

error[E0308]: mismatched types
   --> /build/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/bstr-1.12.1/src/impls.rs:76:47
    |
 76 |                 PartialOrd::partial_cmp(this, other.as_bytes())
    |                 -----------------------       ^^^^^^^^^^^^^^^^ expected `&BStr`, found `&[u8]`
    |                 |
    |                 arguments to this function are incorrect
...
921 |     impl_partial_ord!(&'a BStr, String);
    |     ----------------------------------- in this macro invocation
    |
    = note: expected reference `&BStr`
               found reference `&[u8]`
note: method defined here
   --> library/core/src/cmp.rs:1395:7
    = note: this error originates in the macro `impl_partial_ord` (in Nightly builds, run with -Z macro-backtrace for more info)
help: try removing the method call
    |
 76 -                 PartialOrd::partial_cmp(this, other.as_bytes())
 76 +                 PartialOrd::partial_cmp(this, other)
    |

Some errors have detailed explanations: E0308, E0583, E0599.
For more information about an error, try `rustc --explain E0308`.
error: could not compile `bstr` (lib) due to 34 previous errors
warning: build failed, waiting for other jobs to finish...
```

--- a/extensions/underthesea_core/Cargo.toml
+++ b/extensions/underthesea_core/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "underthesea_core"
-version = "3.3.1"
+version = "3.3.2"
 homepage = "https://github.com/undertheseanlp/underthesea/tree/main/extensions/underthesea_core"
 repository = "https://github.com/undertheseanlp/underthesea/"
 authors = ["Vu Anh <anhv.ict91@gmail.com>"]
@@ -33,7 +33,6 @@ harness = false
 serde = { version = "1.0", features = [ "derive" ] }
 regex = "1"
 rayon = "1.5"
-crfs = "0.1"
 
 # CRF implementation dependencies
 hashbrown = { version = "0.15", features = ["serde"] }  # Fast hash maps with serde support

--- a/extensions/underthesea_core/tests/models.rs
+++ b/extensions/underthesea_core/tests/models.rs
@@ -1,10 +1,13 @@
 #[cfg(test)]
 mod tests {
-    use std::fs;
+    use underthesea_core::crf::serialization::{CRFFormat, ModelLoader};
 
     #[test]
-    fn test_crfs() {
-        let buf = fs::read("tests/wt_crf_2018_09_13.bin").unwrap();
-        let _model = crfs::Model::new(&buf).unwrap();
+    fn test_load_crfsuite_model() {
+        let model = ModelLoader::new()
+            .load("tests/wt_crf_2018_09_13.bin", CRFFormat::Auto)
+            .unwrap();
+        assert!(model.num_labels > 0);
+        assert!(model.num_attributes > 0);
     }
 }

